NuKeeper.AzureDevOps/AzureDevOpsForkFinder.cs

```diff
@@ -11,32 +11,34 @@ public class AzureDevOpsForkFinder : IForkFinder
     {
         private readonly ICollaborationPlatform _collaborationPlatform;
         private readonly INuKeeperLogger _logger;
+        private readonly ForkMode _forkMode;
 
-        public AzureDevOpsForkFinder(ICollaborationPlatform collaborationPlatform, INuKeeperLogger logger)
+        public AzureDevOpsForkFinder(ICollaborationPlatform collaborationPlatform, INuKeeperLogger logger, ForkMode forkMode)
         {
             _collaborationPlatform = collaborationPlatform;
             _logger = logger;
+            _forkMode = forkMode;
         }
 
-        public async Task<ForkData> FindPushFork(ForkMode forkMode, string userName, ForkData fallbackFork)
+        public async Task<ForkData> FindPushFork(string userName, ForkData fallbackFork)
         {
-            _logger.Detailed($"FindPushFork. Fork Mode is {forkMode}");
+            _logger.Detailed($"FindPushFork. Fork Mode is {_forkMode}");
 
-            switch (forkMode)
+            switch (_forkMode)
             {
                 case ForkMode.PreferFork:
                 case ForkMode.PreferSingleRepository:
-                    _logger.Error($"{forkMode} not yet implemented");
+                    _logger.Error($"{_forkMode} not yet implemented");
                     throw new NotImplementedException();
 
                 case ForkMode.SingleRepositoryOnly:
-                    return await FindUpstreamRepoOnly(forkMode, fallbackFork);
+                    return await FindUpstreamRepoOnly(fallbackFork);
 
                 default:
-                    throw new ArgumentOutOfRangeException($"Unknown fork mode: {forkMode}");
+                    throw new ArgumentOutOfRangeException($"Unknown fork mode: {_forkMode}");
```

Comment:
Minor suggestion: move the argument validation to the constructor, and pass the name of the argument as the first parameter of the `ArgumentOutOfRangeException` constructor.